packages/engine/src/avatar/components/AnimationComponent.ts

```diff
@@ -36,3 +42,34 @@ export const AnimationComponent = defineComponent({
     animations: S.Array(S.Type<AnimationClip>())
   })
 })
+
+export const useLoadAnimationFromBatchGLTF = (urls: string[], keepEntities = false) => {
+  const animations = urls.map((url) => useLoadAnimationFromGLTF(url, keepEntities))
+  const loadedAnimations = useHookstate(null as [AnimationClip[] | null, Entity][] | null)
+  useEffect(() => {
+    if (loadedAnimations.value || animations.some((animation) => !animation[0].value)) return
+    loadedAnimations.set(animations.map((animation) => [animation[0].get(NO_PROXY)!, animation[1]]))
+  })
+  return loadedAnimations as State<[AnimationClip[] | null, Entity][]>
+}
+
+export const useLoadAnimationFromGLTF = (url: string, keepEntity = false) => {
+  const assetEntity = useMutableState(GLTFAssetState)[url].value
+  const animation = useHookstate(null as AnimationClip[] | null)
+  const animationComponent = useOptionalComponent(assetEntity, AnimationComponent)
+
+  useEffect(() => {
+    if (animation.value) return
+    if (!assetEntity) {
+      GLTFAssetState.loadScene(url, v4())
+      return
+    }
+  }, [useOptionalComponent(assetEntity, GLTFComponent)?.progress])
```

Comment:
prefer not to use hooks in deps array, we should declare this as a variable like we normally do